docs/rfc/rfc-009-consensus-parameter-upgrades.md

```diff
@@ -0,0 +1,110 @@
+# RFC 009 : Consensus Parameter Upgrade Considerations
+
+## Changelog
+
+- 06-Jan-2011: Initial draft (@williambanfield).
+
+## Abstract
+
+This document discusses the challenges of adding additional consensus parameters
+to Tendermint and proposes a few solutions that can enable addition of consensus
+parameters in a backwards-compatible way.
+
+## Background
+
+This section provides an overview of the issues of adding consensus parameters
+to Tendermint.
+
+### Hash Compatibility
+
+Tendermint produces a hash of a subset of the consensus parameters. The values
```

Comment:
This may be a larger question than this doc wants to tackle, but how can consensus parameters be trusted if they are not included in the signed hashes?

(Even if a parameter value is "optional" in some sense, I would expect the node wants to know that the value presented is agreed-upon)

Reply:
I would assume that their elision from the signed header is just a mistake that we should aim to rectify.